### PR TITLE
{geo}[foss/2022a] ESMF v8.3.0

### PR DESCRIPTION
**Pull request by maxim-masterov**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/e/ESMF/ESMF-8.3.0-foss-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+source_urls = ['https://github.com/esmf-org/esmf/archive/']
+sources = ['v%(version)s.tar.gz']
+patches = ['ESMF-6.1.1_libopts.patch']
+checksums = [
+    '0ff43ede83d1ac6beabd3d5e2a646f7574174b28a48d1b9f2c318a054ba268fd',  # v8.3.0.tar.gz
+    '3851627f07c32a7da55d99072d619942bd3a1d9dd002e1557716158e7aacdaf4',  # ESMF-6.1.1_libopts.patch
+]
+
+# builddependencies = [('libarchive', '3.6.1')]
+
```
Comment (jfgrimm) [inline easybuild/easyconfigs/e/ESMF/ESMF-8.3.0-foss-2022a.eb:20]:
```suggestion
```

**Comment by jfgrimm** [comment]
@boegelbot please test @ generoso

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in total)
node025.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/490a719fe2820731fbef67dbe8896ab6 for a full test report.

**Comment by jfgrimm** [comment]
@maxim-masterov looks like ESMF now needs a `CMake` build dependency
```
14430 tree_lib in: /tmp/jfg508/ESMF/8.3.0/foss-2022a/esmf-8.3.0/src/Infrastructure/IO/PIO                                     
14431 make[6]: Entering directory `/tmp/jfg508/ESMF/8.3.0/foss-2022a/esmf-8.3.0/src/Infrastructure/IO/PIO'                    
14432 makefile:41: warning: overriding recipe for target `tree_lib'                                                           
14433 /tmp/jfg508/ESMF/8.3.0/foss-2022a/esmf-8.3.0//build/common.mk:2278: warning: ignoring old recipe for target `tree_lib'  
14434 cmake --version                                                                                                         
14435 make[6]: cmake: Command not found                                                                                       
14436 make[6]: *** [tree_lib] Error 127
```

**Comment by maxim-masterov** [comment]
@jfgrimm I see, I have cmake installed on both my systems, so I didn't pay attention to where it is taken from. Will add it as a buildependency.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node025.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/be65f1d7a9833ac1787f7450ec679a6a for a full test report.

**Comment by jfgrimm** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
> @maxim-masterov looks like ESMF now needs a `CMake` build dependency
> 
> ```
> 14430 tree_lib in: /tmp/jfg508/ESMF/8.3.0/foss-2022a/esmf-8.3.0/src/Infrastructure/IO/PIO                                     
> 14431 make[6]: Entering directory `/tmp/jfg508/ESMF/8.3.0/foss-2022a/esmf-8.3.0/src/Infrastructure/IO/PIO'                    
> 14432 makefile:41: warning: overriding recipe for target `tree_lib'                                                           
> 14433 /tmp/jfg508/ESMF/8.3.0/foss-2022a/esmf-8.3.0//build/common.mk:2278: warning: ignoring old recipe for target `tree_lib'  
> 14434 cmake --version                                                                                                         
> 14435 make[6]: cmake: Command not found                                                                                       
> 14436 make[6]: *** [tree_lib] Error 127
> ```

In 8.3.0 [they've switched the default](https://github.com/esmf-org/esmf/commit/27baa29b00dcad09c016d567e5c0022a823d3adf) for building the [PIO part](https://github.com/esmf-org/esmf/tree/develop/src/Infrastructure/IO/PIO) to CMake.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node301.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/cb617466e52679e3f0898a00a2788b76 for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @maxim-masterov!

**Pull request merged**